src/features/foldingProvider.ts

```diff
@@ -13,12 +13,52 @@ export default class AsciidocFoldingRangeProvider implements vscode.FoldingRange
   ) {
   }
 
-  public async provideFoldingRanges (
+  public provideFoldingRanges (
     document: vscode.TextDocument,
     _token: vscode.CancellationToken
-  ): Promise<vscode.FoldingRange[]> {
+  ): vscode.FoldingRange[] {
+    const foldingRanges = this.getHeaderFoldingRanges(document)
+    return foldingRanges.concat(AsciidocFoldingRangeProvider.getConditionalFoldingRanges(document))
+  }
+
+  private static getConditionalFoldingRanges (document: vscode.TextDocument) {
+    const conditionalStartRx = /^(\\)?(ifdef|ifndef|ifeval)::(\S*?(?:([,+])\S*?)?)\[(#{CC_ANY}+)?/
+    const conditionalEndRx = /^(\\)?(endif)::(\S*?(?:([,+])\S*?)?)\[(#{CC_ANY}+)?/
```

Comment:
We can move the two regular expressions outside of the class (top of the file after the imports) since we don't need to instantiate them each time this function is called.

We should also add link on: https://github.com/asciidoctor/asciidoctor/blob/0aad7459d1fe548219733b4a2b4f00fd3bf6f362/lib/asciidoctor/rx.rb#L76